### PR TITLE
[flink] Support the creation of tags based on the latest snapshot

### PR DESCRIPTION
**Pull request by MonsterChenzhuo**
### Purpose

Support the creation of tags based on the latest snapshot.

### CODE REVIEW

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/CreateTagActionFactory.java [hunk 1]
```diff
@@ -39,12 +39,19 @@ public String identifier() {
...
     public Optional<Action> create(MultipleParameterToolAdapter params) {
         checkRequiredArgument(params, TAG_NAME);
-        checkRequiredArgument(params, SNAPSHOT);
 
         Tuple3<String, String, String> tablePath = getTablePath(params);
         Map<String, String> catalogConfig = optionalConfigMap(params, CATALOG_CONF);
         String tagName = params.get(TAG_NAME);
-        long snapshot = Long.parseLong(params.get(SNAPSHOT));
+
+        long snapshot = -1;
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/CreateTagActionFactory.java:47]: Long snapshot = null;

File: docs/content/maintenance/manage-tags.md [hunk 2]
```diff
@@ -108,7 +108,7 @@ You can create a tag with given name (cannot be number) and snapshot ID.
     --database <database-name> \ 
     --table <table-name> \
     --tag_name <tag-name> \
-    --snapshot <snapshot-id> \
+    [--snapshot <snapshot_id>] \
```
Comment (JingsongLi) [inline docs/content/maintenance/manage-tags.md:111]: You need to modify Flink and Spark procedure too.

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/CreateTagActionFactory.java (hunk 1 above)
Comment (MonsterChenzhuo) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/CreateTagActionFactory.java]: done.

File: docs/content/maintenance/manage-tags.md (hunk 2 above)
Comment (MonsterChenzhuo) [reply docs/content/maintenance/manage-tags.md:111]: done.

**Comment by MonsterChenzhuo** [comment]
@JingsongLi  PTAL

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**